src/ellipse2d.jl

```diff
@@ -1,16 +1,23 @@
 #Ellipse methods
 
-Ellipse(x::Int, y::Int, ρx::T, ρy::U) where {T<:Real, U<:Real} = Ellipse(Point(x,y), ρx, ρy)
-Ellipse(p::CartesianIndex{2}, ρx::T, ρy::U) where {T<:Real, U<:Real} = Ellipse(Point(p), ρx, ρy)
-Ellipse(circle::CirclePointRadius) = Ellipse(circle.center, circle.ρ, circle.ρ)
+Ellipse(x::Int, y::Int, ρx::T, ρy::U , thickness::Int ,fill::Bool) where {T<:Real, U<:Real} = Ellipse(Point(x,y), ρx, ρy, thickness,fill)
+Ellipse(p::CartesianIndex{2}, ρx::T, ρy::U, thickness::Int ,fill::Bool) where {T<:Real, U<:Real} = Ellipse(Point(p), ρx, ρy,thickness,fill)
+Ellipse(circle::CirclePointRadius) = Ellipse(circle.center, circle.ρ, circle.ρ,circle.thickness,circle.fill)
 
 function draw!(img::AbstractArray{T, 2}, ellipse::Ellipse, color::T) where T<:Colorant
 	ys = Int[]
 	xs = Int[]
+	break_point = 0
+    if ellipse.thickness >= ellipse.ρy || ellipse.thickness >= ellipse.ρx
+			error("Thickness greater than a,b not allowed")
+	end
```

Comment:
and don't mix TAB with 4 SPACES

```suggestion
    if ellipse.thickness >= ellipse.ρy || ellipse.thickness >= ellipse.ρx
        error("Thickness greater than a,b not allowed")
    end
```